toc/rfc/rfc-draft-system-cnb.md

```diff
@@ -0,0 +1,27 @@
+# Meta
+[meta]: #meta
+- Name: System Cloud Native Buildpacks
+- Start Date: 30.04.2024
+- Author(s): @c0d1ngm0nk3y, @pbusko, @nicolasbender, @modulo11
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (fill in with PR link after you submit it)
+- Predecessor: https://github.com/cloudfoundry/community/pull/796
+
+## Summary
+
+This RFC proposes extending the Cloud Foundry functionality to introduce and support system Cloud Native Buildpacks (CNBs). It will allow users to initiate the `cf push --lifecycle cnb` command without providing an explicit list of buildpacks, relying instead on the list of system buildpacks and the built-in auto-detection mechanism.
+
+## Problem
+
+Despite the recent integration of CNBs in the Cloud Foundry platform, the feature still has limitations that could hinder user experience and efficiency. Currently, only custom buildpacks are supported. Users have to manually provide a list of buildpacks (at least one) for their application to run with the `cnb` lifecycle type.
+
+## Proposal
+
+- Introduce a new `lifecycle` column to the [buildpack object](https://v3-apidocs.cloudfoundry.org/version/3.164.0/index.html#the-buildpack-object) with ENUM values `cnb` or `buildpack`, with `buildpack` being the default.
+- Add new parameter `lifecycle` to the [`POST /v3/buildpacks`](https://v3-apidocs.cloudfoundry.org/version/3.164.0/index.html#create-a-buildpack) API endpoint, with the default value being inferred from the `default_app_lifecycle` setting within the Cloud Controller config.
```

Comment:
It is not clear from where the `default_app_lifecycle` comes and I think we should mention it at first place and also its default value. E.g.
```
- Introduce a new global `default_app_lifecycle` property for the Cloud Controller to configure the default... 
```

Reply:
this is an already existing property in CC config: https://github.com/cloudfoundry/capi-release/blob/6a8ac6f03f8a79b8f6ed2556e9c8b469a12634ea/jobs/cloud_controller_ng/templates/cloud_controller_ng.yml.erb#L540

Reply:
Ok, I see.